cmd/snap/cmd_prepare_image.go

```diff
@@ -71,11 +78,20 @@ core device images.
 func (x *cmdPrepareImage) Execute(args []string) error {
 	opts := &image.Options{
 		ModelFile: x.Positional.ModelAssertionFn,
+		Channel:   x.Channel,
+		Snaps:     x.ExtraSnaps,
+	}
 
-		RootDir:         filepath.Join(x.Positional.Rootdir, "image"),
-		GadgetUnpackDir: filepath.Join(x.Positional.Rootdir, "gadget"),
-		Channel:         x.Channel,
-		Snaps:           x.ExtraSnaps,
+	if x.Classic {
+		opts.Classic = true
+		opts.RootDir = x.Positional.Rootdir
```

Comment:
Because we expect prepare-image --classic to be used standalone, and does less, imposes less constraints (no automatic gadget unpacking, no messing with boot env, no kernel extraction), I don't think it makes sense to impose the "/image" structure here. That's also why I made it a mode with an explicit option.